iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PackageInstallationCacheCheck.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.tree.HasTextRange;
+import org.sonar.iac.docker.checks.utils.ArgumentResolutionSplitter;
+import org.sonar.iac.docker.checks.utils.CheckUtils;
+import org.sonar.iac.docker.checks.utils.CommandDetector;
+import org.sonar.iac.docker.checks.utils.command.SeparatedList;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+import static java.util.function.Predicate.not;
+
+@Rule(key = "S6587")
+public class PackageInstallationCacheCheck implements IacCheck {
+
+  private static final String MESSAGE = "Remove cache after installing packages.";
+
+  private static final Set<String> APK_CACHE_LOCATIONS = Set.of("/etc/apk/cache/*", "/var/cache/apk/*");
+  private static final Set<String> APT_COMMANDS = Set.of("apt", "apt-get", "aptitude");
+  private static final Set<String> APT_CACHE_LOCATION = Set.of("/var/lib/apt/lists/*");
+  private static final Predicate<String> containsROrF = s -> s.toLowerCase(Locale.ROOT).contains("r") || s.contains("f");
```

Comment:
When using `RUN apk add nginx && rm /var/cache/apk`, it can result in not actually cleaning the cache, if there are directories inside the apk folder. I don't agree to not raise an issue here, as it's actually still sensitive. What would you suggest? Addming the `rm` as a secondary location?

Reply:
Ok, I assume that there may be cases when check for flags will be valuable. Let's leave the logic as is.

Reply:
TL;DR of the discussion: While it's more a misuse of the `rm` command and a failed attempt at cleaning the cache, we still feel it's valuable to have some the `Remove cache ...` issue displayed in the vicinity. 
As we don't see a lot of misused `rm` commands in dockerfiles, it makes no sense to spend the effort to create secondary locations.